packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -158,6 +158,19 @@ void expectSync(dynamic actual, dynamic matcher, {
   test_package.expect(actual, matcher, reason: reason);
 }
 
+/// Just like [expect], but returns a [Future] that completes when the matcher
+/// has finished matching.
+///
+/// If the matcher fails asynchronously, that failure is piped to the returned
+/// future where it can be handled by user code.
```

Comment:
Explain what happens if it isn't handled by user code (presumably the test fails?)